corehq/warehouse/transforms/sql/synclog_fact.sql

```diff
@@ -0,0 +1,26 @@
+INSERT INTO {{ synclog_fact }} (
+    sync_log_id,
+    sync_date,
+    domain,
+    domain_dim_id,
+    user_id,
+    user_dim_id,
+    build_id,
+    duration,
+    batch_id
+)
+SELECT
+    sync_log_id,
+    sync_date,
+    domain_table.domain,
+    domain_table.id,
+    user_table.user_id,
+    user_table.id,
+    build_id,
+    duration,
+    '{{ batch_id }}'
+FROM
+    {{ synclog_staging }} AS synclog_table
+JOIN {{ domain_dim }} AS domain_table ON synclog_table.domain = domain_table.domain
+-- Allow synclogs to be inserted that do not have users in the UserDim
```

Comment:
why is this necessary. all users should be in the user dim. if they are not that should be seen as a failure, right?

Reply:
I will investigate this a bit more, but I remember I had to do this because of some synclogs with null users.

Reply:
Comes from existing design decision https://github.com/dimagi/commcare-hq/blob/master/corehq/warehouse/transforms/sql/form_fact.sql#L35

But I think we do need a better way to handle